src/main/java/frc/robot/subsystems/climber/Climber.java

```diff
@@ -0,0 +1,117 @@
+package frc.robot.subsystems.climber;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.subsystems.UnitModel;
+
+public class Climber extends SubsystemBase {
+    private final TalonFX master = new TalonFX(Ports.Climber.MASTER);
+    private final TalonFX slave = new TalonFX(Ports.Climber.SLAVE);
+    private final UnitModel unitModel = new UnitModel(Constants.Climber.TICKS_PER_METER);
+    private final Solenoid stopper = new Solenoid(Ports.Climber.STOPPER);
+    private final Solenoid gearboxPiston = new Solenoid(Ports.Climber.GEARBOX_PISTON);

```

Comment:
A couple of things:
1) `gearbox` or `gearShifter`, like you didn't write `stopperPiston`. It's an unintersecting implementation detail.
2) This gearbox and its motors are used by the shooter, too. I think it should be in a separate subsystem. You need to think how you'll make this subsystem (the gearbox) communicate effectively with both the shooter and the climber, maybe by using [suppliers](https://docs.oracle.com/javase/8/docs/api/java/util/function/Supplier.html) and lambda functions. 
I'm not sure that's the best solution for this complicated situation (motors used by multiple subsystems). Consult your teammates and @orel-david.